state/controller.go

```diff
@@ -25,6 +29,47 @@ func controllerKey(controllerUUID string) string {
 	return fmt.Sprintf("%s#%s", controllerGlobalKey, controllerUUID)
 }
 
+// Controller encapsulates state for the Juju controller as a whole,
+// as opposed to model specific functionality.
+type Controller struct {
+	clock                  clock.Clock
+	controllerModelTag     names.ModelTag
+	controllerTag          names.ControllerTag
+	mongoInfo              *mongo.MongoInfo
+	session                *mgo.Session
+	policy                 Policy
+	newPolicy              NewPolicyFunc
+	runTransactionObserver RunTransactionObserverFunc
+}
+
+// Close the connection to the database.
+func (ctlr *Controller) Close() error {
+	ctlr.session.Close()
+	return nil
```

Comment:
FWIW the error was being checked in a bunch of places but of course it was always going to be nil. I have fixed this as well as adding more checks for a returned error.